### DISCUSSION TITLE
Why do segments contain intervening connectors?

### DISCUSSION THREAD

**Discussion by songololo** [OvertureMaps/data#36] [Discussion]
I'm curious about why segments contain intervening connectors.

In OSM a roadway geometry is described by adding points along a line to denote the approximate trajectory of a road. This isn't great because when building a graph the intervening nodes have to be stripped out and a geometry for the line segment has to be created manually. This way a single line segment spans between two intersections.

In Overture the opposite seems to be happening, a single segment can span multiple intersections, so when building a graph it is necessary to split the segment into the smaller intervening segments, and this is quite a bit of work because you first need to split the segment, then you need to track which segments and connectors belong together, and then you can build the graph.

Curious why Overture doesn't simply have a 1-to-2 relationship between segments and connectors?

**Comment by brad-richardson** [comment]
@songololo Great question, sorry for the delay here. A tradeoff we're making is to have more usable stable IDs over longer segments, enabled by multiple connectors and geometrically-scoped attributes. As this pattern isn't compatible with all systems and isn't how OSM operates, we've been developing a tool that can "split" segments into the 1-to-2 connector output ratio you're describing. We're hoping to open source that soon. cc: @ibnt1

**Reply by brad-richardson** [reply]
We've now published our open source splitter that allows for creating this 2:1 relationship using the default configuration: https://github.com/OvertureMaps/transportation-splitter. Please let us know if that doesn't meet your needs or how we can improve the tool.

**Reply by songololo** [reply]
Thanks for the update
